deltacat/storage/model/partition_spec.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+from typing import List, Optional, Any
+from deltacat.storage.model.transform import Transform
+
+"""
+An ordered list of partition values determining the values of
+ordered transforms specified in the partition spec.
+"""
+PartitionValues = List[Any]
+
+
+class PartitionFilter(dict):
+    """
+    This class represents a filter for partitions.
+    It is used to filter partitions based on certain criteria.
+    """
+
+    @staticmethod
+    def of(
+        partition_values: Optional[PartitionValues] = None,
+    ) -> PartitionFilter:
+        """
+        Creates a new PartitionFilter instance with the specified partition key and value.
+        """
+        partition_filter = PartitionFilter()
+        partition_filter["partitionValues"] = partition_values
+        return partition_filter
+
+    @property
+    def partition_values(self) -> Optional[PartitionValues]:
+        return self.get("partitionValues")
+
+
+class PartitionSpec(dict):
```

Comment:
To better differentiate this from a strict Iceberg PartitionSpec and make for a more straightforward future merge with the similar proposed model changes from https://github.com/pdames/deltacat/blob/iceberg/deltacat/storage/model/partition.py, I'd recommend just calling this `PartitionScheme` and including it in the existing `partition.py` together with other classes related to partitioning here.

I also think we should take the existing model there of attaching the transform to a partition key with a name and distinct ID vs. defining the spec directly as an ordered list of transforms.